### PR TITLE
Script for automating build process: download & verify & build & upload

### PR DESCRIPTION
**Pull request by marmarek**
Implements major part of QubesOS/qubes-issues#1818

### CODE REVIEW

**Comment by marmarek** [comment]
This (and https://github.com/QubesOS/qubes-builder-github/issues/7) could really use some review.
/cc @woju @jpouellet 
This is currently a blocker for setting up a builder for Qubes 4.0 and also for https://github.com/qubesos-contrib.

File: doc/Configuration.md [hunk 1]
```diff
@@ -248,6 +248,33 @@ Public self-signed certificate (it will be generated if doesn't exists)
...
+### GITHUB_BUILD_ISSUES_REPO
+> Default: no value
+
+Github repository name (in for of `owner/name`) where build failures will be
+reported (as github issues).
+
+### GITHUB_BUILD_REPORT_REPO
+> Default: no value
+
+Github repository name (in for of `owner/name`) where all finished builds will
```
Comment (jpouellet) [inline doc/Configuration.md]: s/for of/form/

File: Makefile [hunk 2]
```diff
@@ -637,7 +635,19 @@ do-merge:
 	for REPO in $$REPOS; do \
 		git -C $$REPO rev-parse -q --verify FETCH_HEAD >/dev/null || continue; \
 		echo "Merging FETCH_HEAD into $$REPO"; \
-		git -C $$REPO merge --ff --no-edit FETCH_HEAD || exit 1; \
+		git -C $$REPO merge --ff $(GIT_MERGE_OPTS) --no-edit FETCH_HEAD || exit 1; \
+	done
+
+do-merge-versions-only:
+	@set -a; \
```
Comment (jpouellet) [inline Makefile]: Why `set -a`? Does some command here use REPOS or components_var as env var?

Comment (marmarek) [inline Makefile]: I don't see any, just a copy&paste - prepare-merge do make use of it.

**Comment by marmarek** [comment]
Applied changes suggested by @woju and @jpouellet

**Comment by jpouellet** [comment]
Do not interpret the presence and resolution of a suggestion from me as passing a meaningful review from me. I did not spend time to properly audit this.

**Comment by marmarek** [comment]
@jpouellet Are you planning to? If so, I can wait for it. But since at least one other person already have reviewed it (@woju), I think it isn't that important anymore.

**Comment by jpouellet** [comment]
I'd hoped to get to this by now, but am (very slowly) making my way through the rest of qubes-builder in order to properly understand the full context first.

If you feel confident it is okay and this is blocking significant progress in other areas, then just go ahead. Free time is unfortunately not as abundant as I would like...

**Comment by jpouellet** [comment]
Finally got around to reviewing this.

See comments on the recent commits in qubes-builder and qubes-builder-github.

**Pull request merged**